greenmail-core/src/main/java/com/icegreen/greenmail/filestore/MailboxEntries.java

```diff
@@ -0,0 +1,181 @@
+package com.icegreen.greenmail.filestore;
+
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;
+import static java.nio.file.StandardOpenOption.WRITE;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+
+import com.icegreen.greenmail.filestore.fs.MessageToFS;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Binary file which contains some information for each message in a mailbox.
+ * File name is normally greenmail.messageEntries.binary
+ * See methods writeToDOS() and readFromDIS() for more information about the content of the file.
+ */
+class MailboxEntries {
+
+    final Logger log = LoggerFactory.getLogger(FileHierarchicalFolder.class);
+
+    final Object syncLock = new Object();
+
+    // Mailbox list which are stored in the mailboxEntriesFile, every change here must be synced to the FS:
+    ArrayList<MessageEntry> list = new ArrayList<>();
+
+    private Path mailboxEntriesFile;
+
+    MailboxEntries(Path pathToEntriesFile) {
+        this.mailboxEntriesFile = pathToEntriesFile;
+    }
+
+    void storeFileToFS() {
```

Comment:
Why does this method exist? It just calls the other method and puts a sync around it